src/replication.c

```diff
@@ -1731,6 +1731,20 @@ void readSyncBulkPayload(connection *conn) {
          * dictionaries. */
         diskless_load_backup = disklessLoadMakeBackup();
     }
+
+    /* Replica starts to appply data from new master, we must discard the cached
+     * master structure. And before this, make sure there is no master client,
+     * since replica's data would be changed and cached master is exactly wrong
+     * once creating master client. */
```

Comment:
what did you mean by
> since replica's data would be changed and cached master is **exactly wrong once** creating master client

Reply:
I think i should say 
>since replica's data would be changed and cached master is exactly wrong once the new master client is created.

If the cached master is existed, this means the redis dataset still is not changed from the time of disconnecting with old master. But if the master client is created, this means the redis starts to apply data from new master, it is wrong if there still is a cached master. Because we delay to discard cached master, so i want to make sure there is no master client before discarding cached master.